conditions.go

```diff
@@ -75,6 +75,60 @@ func (c Condition) Validate() error {
 	return nil
 }
 
+func (c Condition) MarshalJSON() ([]byte, error) {
+	var serialized string
+	if c != nil {
+		serialized = c.String()
+	}
+	return json.Marshal(fmt.Sprintf("cond:%s", serialized))
```

Comment:
This looks much better.

I wonder if we don't want to just drop the hex version and use no prefix.
We have no legacy addresses in genesis files or external usages, as this is the first time we expose conditions serialized, so we can avoid the prefix complexity of addresses and just `json.Marshal(serialized)`

Reply:
I wanted to be consistent with address prefix pattern but I agree no need for this. Good point! 🎉